drink/src/session.rs

```diff
@@ -395,3 +470,71 @@ impl<R: Runtime> Session<R> {
         self.sandbox.override_debug_handle(d);
     }
 }
+
+/// A struct representing the result of parsing a `.contract` bundle file.
+///
+/// It can be used with the following methods of the `Session` struct:
+/// - `deploy_bundle`
+/// - `deploy_bundle_and`
+/// - `upload_bundle`
+/// - `upload_bundle_and`
+pub struct ContractBundle {
+    /// WASM blob of the contract
+    bytes: Vec<u8>,
+    /// Transcoder derived from the ABI/metadata
+    transcoder: Rc<ContractMessageTranscoder>,
+}
+
+impl ContractBundle {
+    /// Load and parse the information in a `.contract` bundle under `path`, producing a `ContractBundle` struct.
+    pub fn load<P>(path: P) -> Result<Self, SessionError>
+    where
+        P: AsRef<std::path::Path>,
+    {
+        let metadata: ContractMetadata = ContractMetadata::load(&path).map_err(|e| {
+            SessionError::BundleLoadFailed(format!("Failed to load the contract file:\n{e:?}"))
+        })?;
+
+        let ink_metadata = serde_json::from_value(serde_json::Value::Object(metadata.abi))
+            .map_err(|e| {
+                SessionError::BundleLoadFailed(format!(
+                    "Failed to parse metadata from the contract file:\n{e:?}"
+                ))
+            })?;
+
+        let bytes = metadata
+            .source
+            .wasm
+            .ok_or(SessionError::BundleLoadFailed(
+                "Failed to get the WASM blob from the contract file".to_string(),
+            ))?
+            .0;
+
+        let transcoder = Rc::new(ContractMessageTranscoder::new(ink_metadata));
+
+        Ok(Self { bytes, transcoder })
+    }
+
+    /// Load the `.contract` bundle (`bundle_file`) located in the `project_dir`` working directory.
+    ///
+    /// This is meant to be used predominantly by the `local_bundle!` macro.
+    pub fn local(project_dir: &str, bundle_file: String) -> Self {
```

Comment:
this one doesn't have to be public, does it? we don't advertise it as a recommended API for the end user, so maybe we can make it private

Reply:
oh, it has to be public, because of macro, right?

Reply:
@deuszx the reason I went with `bundle` instead of `contract` (especially in the name of the struct but here as well) was to make it very clear we're not talking about some deployed contract instance but just the result of parsing the `.contract` file. But maybe the name `ContractFile` would be also better than `ContractBundle`?

Reply:
@pmikolajczyk41 yeah, that was my reasoning here :)

Reply:
> @deuszx the reason I went with `bundle` instead of `contract` (especially in the name of the struct but here as well) was to make it very clear we're not talking about some deployed contract instance but just the result of parsing the `.contract` file. But maybe the name `ContractFile` would be also better than `ContractBundle`?

`ContractBundle` for the struct is OK, I was only worried about the arg's name.